### ISSUE TITLE
condition in filters.assign returning an error in python

### ISSUE THREAD

**Issue by spiegelss** [PDAL/PDAL#2819]
**Describe the bug**
Good afternoon,
I am attempting to assign values to points using  **filters.assign** and **filters.overlay**.  According to the [documentation for filters.assign](https://pdal.io/stages/filters.assign.html#filters-assign), the condition statement is a list of ranges outlined in [filters.range](https://pdal.io/stages/filters.range.html#ranges).  Here are two examples pipelines and the corresponding outputs.  

**Output 1**
```python
import pdal
import json
pipe = {
    'pipeline': [
          {
    'type': 'readers.las',
   'filename': '/some/file.las'
           },
          {
     'type': 'filters.range',
   'limits': 'X[0:22],Y[0:45]'
          },
         {
         'type': 'writers.las',
          'filename': '/some/file-output.las'
          }
]
}
pipeline = pdal.Pipeline(json.dumps(pipe))
pipeline.validate()


>>>True
```

**Output 2**
```python
import pdal
import json
pipe = {
    'pipeline': [
          {
    'type': 'readers.las',
   'filename': '/some/file.las'
           },
          {
    'type': 'filters.assign',
    'assignment': 'Classification[:]=17',
    'condition': 'X[0:22],Y[0:45]'
         },
         {
         'type': 'writers.las',
          'filename':  '/some/file-output.las'
          }
]
}
pipeline = pdal.Pipeline(json.dumps(pipe))
pipeline.validate()


>>> RuntimeError: Invalid characters following valid range.
```
This leads me to believe **filters.assign** doesn't expect another value after the first condition.  If that is the case, how does one make multiple conditions?  Thank you!

**System/installation information:**
PDAL 2.0.0 with Docker on Ubuntu 18.04

**Comment by abellgithub** [comment]
The documentation is incorrect.  The `condition` options currently accepts only a single range on which to filter.

**Comment by spiegelss** [comment]
Ok gotcha.  May I make this a feature request if it is not already?  In the meantime I can do this:
```python
import pdal
import json
pipe = {
    'pipeline': [
          {
    'type': 'readers.las',
   'filename': '/some/file.las'
           },
          {
     'type': 'filters.range',
   'limits': 'X[0:22],Y[0:45]' # or whatever range
          },
         {
          'type':'filters.assign',
          'assignment':'Classification[:]=3' #or whatever classification
         },
         {
         'type': 'writers.las',
          'filename': '/some/file-output-name.las'
          }
]
}
pipeline = pdal.Pipeline(json.dumps(pipe))
pipeline.validate()
pipeline.execute
```
I'm going to run this in a loop and then combine all the Las files using **filters.merge**, which seems to work.

**Comment by abellgithub** [comment]
You can combine a bunch of LAS files into one by simply running `pdal merge`.

**Comment by abellgithub** [comment]
Closed for #2825
